src/collective/volto/sitesettings/restapi/deserializer/controlpanels.py

```diff
@@ -86,7 +90,27 @@ def __call__(self):
             if errors:
                 raise BadRequest(errors)
 
-        # set width and height for logo and favicon
+        self.align_site_title(data=data)
+        self.set_logo_and_favicon_sizes(data=data)
+
+    def align_site_title(self, data):
+        """align site_title with custom one"""
+        site_title_translated = data.get("site_title_translated", "")
+
+        if site_title_translated:
+            site_title_translated = json.loads(site_title_translated)
+        else:
+            site_title_translated = {}
+
+        lang = api.portal.get_default_language()
+        new_site_title = site_title_translated.get(lang, "")
+
+        if new_site_title:
+            proxy = self.registry.forInterface(ISiteSchema, prefix=self.schema_prefix)
+            setattr(proxy, "site_title", new_site_title.replace("\n", " "))
```

Comment:
what if a site has multiple language (multilingual)? are you changing/writing on the db each time ... ?